commons-hpcc/src/main/java/org/hpccsystems/commons/errors/HpccError.java

```diff
@@ -0,0 +1,321 @@
+     *            the unique Identifier of the object that caused this error
+     */
+    public void setSourceID(String sourceID)
+    {
+        this.sourceID = sourceID;
+    }
+
+    /**
+     * @return the Error Code of this Error
+     */
+    public IErrorCode getErrorCode()
+    {
+        return errorCode;
+    }
+
+    /**
+     * @return the Error message of this error
+     */
+    public String getErrorString()
+    {
+        return errorString;
+    }
+
+    /**
+     * @return The Line Number this error occurred at
+     */
+    public Integer getLineNum()
+    {
+        return linenum;
+    }
+
+    /**
+     * @param l
+     *            Set the line number this error occurred at
+     */
+    public void setLineNum(int l)
+    {
+        linenum = l;
+    }
+
+    /**
+     * @return the Column Number this error occurred at
+     */
+    public Integer getColNum()
+    {
+        return colnum;
+    }
+
+    /**
+     * @param c
+     *            the Column Number this error occurred at
+     */
+    public void setColNum(int c)
+    {
+        colnum = c;
+    }
+
+    /**
+     * @return The Filename of the file this error occurred in
+     */
+    public String getFilename()
+    {
+        return filename;
+    }
+
+    /**
+     * @param filename
+     *            the Filename of the file this error occurred in
+     */
+    public void setFilename(String filename)
+    {
+        this.filename = filename;
+    }
+
+    /**
+     * @param val
+     *            the Source of this object (Composition, Contract, Kel file, etc)
+     */
+    public void setErrorSource(IErrorSource val)
+    {
+        this.errorSource = val;
+    }
+
+    /**
+     * @return the Source of this error (Composition, Contract, Kel , etc)
+     */
+    public IErrorSource getErrorSource()
+    {
+        return errorSource;
+    }
+
+    /**
+     * @return the Type of error: VALIDATE, SYSTEM, HPCC, etc. Class must extend IErrorType
+     */
+    public IErrorType getErrorType()
+    {
+        return errorType;
+    }
+
+    /**
+     * @param val
+     *            the Type of error: VALIDATE, SYSTEM, HPCC, etc. Class must extend IErrorType
+     */
+    public void setErrorType(IErrorType val)
+    {
+        this.errorType = val;
+    }
+
+    /**
+     * @return any additional info for this exception
+     */
+    public String getAdditionalInfo()
+    {
+        return additionalInfo;
+    }
+
+    /**
+     * @param info
+     *            any additional info for this exception
+     */
+    public void setAdditionalInfo(String info)
+    {
+        this.additionalInfo = info;
+    }
+
+    /**
+     * @return Error Level (FATAL, ERROR, WARNING, INFO)
+     */
+    public HpccErrorLevel getErrorLevel()
+    {
+        return errorLevel;
+    }
+
+    /*
+     * (non-Javadoc)
+     * 
+     * @see java.lang.Object#hashCode()
+     */
+    public int hashCode()
+    {
+        return this.toEclIdeString().hashCode();

```

Comment:
This was copied over from the tardis ErrorBlock code--in that codebase, we overrode hashcode to prevent completely duplicate errors from existing multiple times in an errorblock. (If two errors had the same tostring, the are considered "identical". ) In this case, if the tostring for two errors both have "unknown file"  + the same error message they'd be the same hashcode; but if their error message was different it'd be different.

The hashCode override to ensure error uniqueness in an error block is tbh a bit of a hack and I understand if you'd rather have it removed from this codebase.